### PR TITLE
fallback to fetching source control details for default github trigger

### PR DESCRIPTION
**Pull request by ReallyLiri**
In default Google Cloud Build Github triggers, the `build` object structure is a bit different, and fetching the source control info requires some different fields.

example fields from custom and standard trigger:
```
{ 
    id: 'xxx', 
    projectId: 'xxx', 
    status: 'FAILURE', 
    source: { repoSource: { projectId: 'testP', repoName: 'tesRt', branchName: 'master' } }, 
    steps: [ ... ], 
    results: { ... }, 
    createTime: '2019-08-06T05:58:09.900770897Z', 
    startTime: '2019-08-06T05:58:10.810848454Z', 
    finishTime: '2019-08-06T06:02:39.092998458Z', 
    timeout: '600s', 
    logsBucket: 'gs://xxx', 
    sourceProvenance: { resolvedRepoSource: { projectId: 'testP', repoName: 'testR', commitSha: '...' } }, 
    buildTriggerId: 'xxx', 
    options: { substitutionOption: 'ALLOW_LOOSE', logging: 'LEGACY' }, 
    logUrl: 'https://console.cloud.google.com/gcr/builds/xxx', 
    tags: [ 'trigger-xxx' ], 
    timing: { ... } 
}
```

and from the default github trigger:
```
{ 
    id: 'xxx', 
    projectId: 'xxx', 
    status: 'FAILURE', 
    source: { storageSource: { bucket: '...', object: 'xxx.tar.gz' } }, 
    steps: [ ... ], 
    results: { ... }, 
    createTime: '2019-08-06T07:11:55.055137253Z', 
    startTime: '2019-08-06T07:11:56.320654777Z', 
    finishTime: '2019-08-06T07:19:04.865592Z', 
    timeout: '600s', 
    logsBucket: 'gs://xxx', 
    sourceProvenance: { resolvedStorageSource: { bucket: '...', object: 'xxx.tar.gz', generation: 'xxx' }, fileHashes: { 'gs://xxx': [Object] } }, 
    buildTriggerId: 'default-github-checks', 
    options: { substitutionOption: 'ALLOW_LOOSE', logging: 'LEGACY' }, 
    logUrl: 'https://console.cloud.google.com/gcr/builds/xxx', 
    substitutions: { BRANCH_NAME: 'testB', COMMIT_SHA: '...', REPO_NAME: 'testR', REVISION_ID: '....', SHORT_SHA: '...', TAG_NAME: '' }, 
    tags: [ 'trigger-default-github-checks' ], 
    timing: { ... } 
}
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -108,9 +108,18 @@ module.exports.createSlackMessage = async (build, githubCommit) => {
     ],
   };
 
+  var repoName, branchName;
```
Comment (naseemkullah) [inline index.js]: I think `let` is preferable to `var`?

File: index.js [hunk 2]
```diff
@@ -108,9 +108,18 @@ module.exports.createSlackMessage = async (build, githubCommit) => {
...
+  if (build.source && build.source.repoSource) {
+    repoName = build.source.repoSource.repoName;
+    branchName = build.source.repoSource.branchName;
```
Comment (naseemkullah) [inline index.js]: According to the linter I use, it is preferrable to use destructuring e.g.

```
{ repoName } = build.source.repoSource;
{ branchName } = build.source.repoSource;
```

File: index.js [hunk 3]
```diff
@@ -108,9 +108,18 @@ module.exports.createSlackMessage = async (build, githubCommit) => {
...
+    branchName = build.substitutions.BRANCH_NAME;
+  }
+
   // Add source information to the message.
-  const { repoSource } = build.source || null;
-  if (repoSource) {
+  if (repoName && branchName) {
     message.attachments[0].fields.push({
       title: 'Repository',
       value: repoSource.repoName,
```
Comment (naseemkullah) [inline index.js]: I think it should just be `repoName` now, not `repoSource.repoName` ?

Comment (naseemkullah) [reply index.js]: and similarly, further below at line 130, it should be simply `value: branchName`, right?

**Comment by naseemkullah** [comment]
@Philmod could you please double check my comments? Thanks.

**Comment by naseemkullah** [comment]
@ReallyLiri @Philmod can we go over my suggested changes and get this one in? `substitutions` is indeed the way to get the repo and branch names from github app triggered builds

**Comment by naseemkullah** [comment]
cc @liiri

**Comment by Philmod** [comment]
@ReallyLiri Could you also add unit testing?

**Review by naseemkullah — APPROVED** [review]

**Comment by naseemkullah** [comment]
bump

**Comment by ReallyLiri** [comment]
@naseemkullah checks haven't completed, cannot merge

**Comment by Philmod** [comment]
Please merge `master` branch back in your branch. I also fixed the build, it should be triggered by that change.

**Comment by ReallyLiri** [comment]
> Please merge `master` branch back in your branch. I also fixed the build, it should be triggered by that change.

Done

**Pull request merged**